src/fractal_faim_hcs/MetaSeriesUtils_dask.py

```diff
@@ -0,0 +1,302 @@
+import re
+from pathlib import Path
+from typing import Callable
+
+import dask.array as da
+import numpy as np
+import pandas as pd
+from faim_hcs.MetaSeriesUtils import (
+    _build_ch_metadata,
+    compute_z_sampling,
+    get_img_YX,
+    montage_grid_image_YX,
+)
+from faim_hcs.UIntHistogram import UIntHistogram
+from numpy._typing import ArrayLike
+from tifffile import tifffile
+
+
+
+def get_well_image_CZYX_lazy(  # noqa: C901
+    well_files: pd.DataFrame,
+    channels: list[str],
+    assemble_fn: Callable = montage_grid_image_YX,
+) -> tuple[ArrayLike, list[UIntHistogram], list[dict], dict]:
+    """Assemble image data for the given well-files."""
+    # TODO: This function assumes that there are no gaps in data (i.e. that
+    # there are no images missing to fill a full FCZ-stack)
+    # -> maybe implement checks
+
+    planes = sorted(well_files["z"].unique(), key=int)
+
+    fields = sorted(
+        well_files["field"].unique(),
+        key=lambda s: int(re.findall(r"(\d+)", s)[0])
+    )
+
+    # Create an empty np array to store the filenames in the correct structure
+    fn_dtype = f"<U{max([len(fn) for fn in well_files['path']])}"
+    fns_np = np.zeros(
```

Comment:
The creation of fns_np could be abstracted into its own function that just returns the fns_np.